### PR TITLE
[3/n] Setup CamperAPIClient and trivial fixes

### PR DESCRIPTION
**Pull request by e-wai**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)

## Implementation description
* 

## Steps to test
1.

## What should reviewers focus on?
* 

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/AdditionalInfo/QuestionCards/CamperQuestionsCard.tsx [hunk 1]
```diff
@@ -29,9 +29,9 @@ const CamperQuestionsCard = ({
     <QuestionsCardWrapper title={`${camper.firstName} ${camper.lastName}`}>
       <VStack py="24px">
         <Wrap width="100%" px="20px" justify="space-between">
-          {campSpecificFormQuestions.map((question) => (
+          {campSpecificFormQuestions.map((question, questionIndex) => (
```
Comment (dhmi19) [inline frontend/src/components/pages/RegistrantExperience/RegistrationSteps/AdditionalInfo/QuestionCards/CamperQuestionsCard.tsx:32]: tejas updated this in his bug fix PR (which I think we might merge before this PR). You might need to update this later due to merge conflicts

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/AdditionalInfo/additionalInfoReducer.tsx [hunk 2]
```diff
@@ -53,7 +53,10 @@ export const checkAdditionalQuestionsAnswered = (
   const requiredQuestions = campSpecificFormQuestions
     .filter((question) => question.required)
     .map((question) => question.question);
+
   return campers.every((camper) =>
-    requiredQuestions.every((question) => camper.formResponses?.get(question)),
+    requiredQuestions.every(
+      (question) => camper.formResponses?.get(question) ?? false,
```
Comment (dhmi19) [inline frontend/src/components/pages/RegistrantExperience/RegistrationSteps/AdditionalInfo/additionalInfoReducer.tsx:59]: why do we need the explicit `??` check here

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/PaymentSummary.tsx [hunk 3]
```diff
@@ -96,7 +94,7 @@ const PaymentSummary = ({
           <Tbody>
             {items.map((item, index) => (
               <Tr
-                key={item.name}
+                key={index}
```
Comment (dhmi19) [inline frontend/src/components/pages/RegistrantExperience/RegistrationSteps/ReviewRegistration/PaymentSummary.tsx:97]: can we make keys more descriptive? There is a greater chance of overlapping keys if we keep making them numbers

**Review by dhmi19 — APPROVED** [review]
Left some comments but nothing major. good to merge whenever you feel confident

File: frontend/src/components/pages/RegistrantExperience/RegistrationSteps/AdditionalInfo/additionalInfoReducer.tsx (hunk 2 above)
Comment (e-wai) [reply frontend/src/components/pages/RegistrantExperience/RegistrationSteps/AdditionalInfo/additionalInfoReducer.tsx:59]: Wanted to clarify the behaviour with `undefined`, it should fail

**Pull request merged**